Update [ScoreCard] component (docs & examples)

manojleaton opened the pull request:
<!-- If this pull request fixes an Issue, link it below. If not, you can remove the line below -->
Fixes #BLUI-4374 .

<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:
- Updated ScoreCard component with MD3Theme.
- Please verify this PR for kitchen sink example updates for ScoreCard component [ScoreCard KitchenSink](https://github.com/etn-ccis/blui-react-native-showcase-demo/pull/90)

<!-- Include screenshots if they will help illustrate the changes in this PR -->
#### Screenshots / Screen Recording (if applicable)
<img width="328" alt="Screenshot 2023-10-16 at 2 36 12 PM" src="https://github.com/etn-ccis/blui-react-native-component-library/assets/130662346/c105dff6-1b57-48e1-b860-a1d0c488255a">



<!-- Instruction for PR reviewers, if more complicated than a simple yarn start -->
#### To Test:
- yarn start

<!-- Useful for draft pull requests -->
#### Any specific feedback you are looking for?
-

ArshdeepSingh3-eaton commented on components/src/md3/ScoreCard/ScoreCard.tsx:
```diff
@@ -0,0 +1,412 @@
+    ImageStyle,
+    TextStyle,
+} from 'react-native';
+import * as Typography from '../../core/typography';
```
Replace Typography with Text Component

ArshdeepSingh3-eaton replied:
Please Make sure to Not import anything from core folder

ektaghag-eaton reviewed (commented):
1. Fontsize for title, subtitle, and info doesn't match with [Figma](https://www.figma.com/file/Nydz8n8HBj6U3ZtULja3gD/%F0%9F%92%AB-Brightlayer-UI-Tokens-%2B-Components-(in-Material-Design-3)?type=design&node-id=777-18275&mode=design&t=gNVKKVtLy2vtjZIh-0)
2. ScoreCard Header size is 112px currently it is 100px as per Figma
3. Icon size breaks when passing iconSize={48}
<img width="334" alt="Screenshot 2023-10-13 at 1 30 14 PM" src="https://github.com/etn-ccis/blui-react-native-showcase-demo/assets/133877691/a20296b0-6e82-4e16-89aa-1339cff1449b">

We need to test the Score Card again once InfolistItem and HeroBanner are available in the MD3 Theme

huayunh commented:
the corner radius of the four corners are different

And please give it a default background color that is not pure white. I recommend "surface container low"

huayunh commented:
Please update the header background color to use that "primary container" color by default

![image](https://github.com/etn-ccis/blui-react-native-component-library/assets/8997218/c5c61479-1940-44b3-8083-7f52acfcce12)

manojleaton commented:
Added space between Buttons and Stepper as mentioned in the demo. I have updated that change in this PR itself since it's a minor change. Please verify.
<img width="331" alt="Screenshot 2023-10-16 at 5 23 09 PM" src="https://github.com/etn-ccis/blui-react-native-component-library/assets/130662346/6e9f69fc-e607-4f0c-8629-e8b8a9720d8d">

huayunh commented:
I played with it on the design side and actually found 24px to be more comfortable
Do you mind making it 24 instead?

Here's what it would look like on the design side:

![image](https://github.com/etn-ccis/blui-react-native-component-library/assets/8997218/c7966049-a55a-4a05-a750-8bdcd0d071be)

manojleaton commented:
> I played with it on the design side and actually found 24px to be more comfortable Do you mind making it 24 instead?
> 
> Here's what it would look like on the design side:
> 
> ![image](https://user-images.githubusercontent.com/8997218/275532874-c7966049-a55a-4a05-a750-8bdcd0d071be.png)

have added 24px padding, please check

<img width="320" alt="Screenshot 2023-10-16 at 8 44 20 PM" src="https://github.com/etn-ccis/blui-react-native-component-library/assets/130662346/92cc58ef-b9b3-4da7-ae24-985e4fcfff22">

huayunh commented:
And can you adjust the scorecard header padding too? I think this was redesigned a bit to fit with the new typography. 

The vertical padding is now 24; horizontal padding shall stay at 16

joebochill commented:
It looks like something strange is happening with the padding/margin between the stepper and the buttons...your screenshot when you had it set to 16 it looked like 8, when you had it set to 24 it looks like 16...is there some overlap happening?

joebochill commented:
I think the issue with the MobileStepper is more an issue with overflow. The flexing of the containers and the spacing looks to be working as expected (at least on my machine), but the dots are just overflowing out of the container when there are too many of them, making it look like the spacing is broken.

I think the more robust solution would be to make `stepperContainer` flex:0 when the mode is text or dots and flex: 1 when the mode is progress. There will still be issues with overflow if there are too many dots (not new), but the flex behavior will I think be a bit more appropriate (we can still remove the flex wrapper around the buttons in the example, but it should also work if the flex wrapper IS there). What do you think?

manojleaton commented:
> I think the issue with the MobileStepper is more an issue with overflow. The flexing of the containers and the spacing looks to be working as expected (at least on my machine), but the dots are just overflowing out of the container when there are too many of them, making it look like the spacing is broken.
> 
> I think the more robust solution would be to make `stepperContainer` flex:0 when the mode is text or dots and flex: 1 when the mode is progress. There will still be issues with overflow if there are too many dots (not new), but the flex behavior will I think be a bit more appropriate (we can still remove the flex wrapper around the buttons in the example, but it should also work if the flex wrapper IS there). What do you think?

Please refer this PR for the changes as per comment [MobileStepper](https://github.com/etn-ccis/blui-react-native-component-library/pull/415)

ektaghag-eaton commented on components/src/md3/Hero/Hero.tsx:
```diff
@@ -29,8 +29,8 @@ const makeStyles = (theme: MD3Theme, fontScale: number): StyleSheet.NamedStyles<
             overflow: 'hidden',
-            height: 36 * fontScale,
-            width: 36 * fontScale,
+            // height: typeof iconSize === 'number' ? normalizeIconSize() : iconSize * fontScale,
```
Please remove commented code

JeffGreiner-eaton reviewed (approved):
looks good, just Ekta's comment
